tests/components/august/test_init.py

```diff
@@ -1,136 +1,146 @@
-            RequestException("mocked request error"),
-        ]
+    lock_abc_name = hass.states.get("lock.abc_name")
+    assert lock_abc_name is None
+    lock_a6697750d607098bae8d6baa11ef8063_name = hass.states.get(
+        "lock.a6697750d607098bae8d6baa11ef8063_name"
     )
+    assert lock_a6697750d607098bae8d6baa11ef8063_name.state == STATE_LOCKED
 
-    assert data.lock_has_doorsense("doorsenselock1") is True
-    assert data.lock_has_doorsense("nodoorsenselock1") is False
-
-    # The api calls are mocked to fail on the second
-    # run of async_get_lock_detail
-    #
-    # This will be switched to await data.async_get_lock_detail("doorsenselock1")
-    # once we mock the full home assistant setup
-    data._update_locks_detail()
-    # doorsenselock1 should be false if we cannot tell due
-    # to an api error
-    assert data.lock_has_doorsense("doorsenselock1") is False
 
+async def test_lock_has_doorsense(hass):
+    """Check to see if a lock has doorsense."""
+    doorsenselock = await _mock_doorsense_enabled_august_lock_detail(hass)
+    nodoorsenselock = await _mock_doorsense_missing_august_lock_detail(hass)
+    await _create_august_with_devices(hass, [doorsenselock, nodoorsenselock])
 
-async def test__refresh_access_token(hass):
-    """Test refresh of the access token."""
-    authentication = _mock_august_authentication("original_token", 1234)
-    authenticator = _mock_august_authenticator()
-    token_refresh_lock = asyncio.Lock()
-
-    data = august.AugustData(
-        hass, MagicMock(name="api"), authentication, authenticator, token_refresh_lock
-    )
-    await data._async_refresh_access_token_if_needed()
-    authenticator.refresh_access_token.assert_not_called()
-
-    authenticator.should_refresh.return_value = 1
-    authenticator.refresh_access_token.return_value = _mock_august_authentication(
-        "new_token", 5678
+    binary_sensor_online_with_doorsense_name_open = hass.states.get(
+        "binary_sensor.online_with_doorsense_name_open"
     )
-    await data._async_refresh_access_token_if_needed()
-    authenticator.refresh_access_token.assert_called()
-    assert data._access_token == "new_token"
-    assert data._access_token_expires == 5678
-
-
-def _create_august_data_with_lock_details(lock_details):
-    locks = []
-    for lock in lock_details:
-        if isinstance(lock, LockDetail):
-            locks.append(_mock_august_lock(lock.device_id))
-    authentication = _mock_august_authentication("original_token", 1234)
-    authenticator = _mock_august_authenticator()
-    token_refresh_lock = MagicMock()
-    api = MagicMock()
-    api.get_lock_detail = MagicMock(side_effect=lock_details)
-    api.get_operable_locks = MagicMock(return_value=locks)
-    api.get_doorbells = MagicMock(return_value=[])
-    return august.AugustData(
-        MagicMock(), api, authentication, authenticator, token_refresh_lock
+    assert binary_sensor_online_with_doorsense_name_open.state == STATE_ON
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5068548)